File: 122/js/solution.js
var maxProfit = function (prices) {
  if (prices.length <= 0) return 0;
  let profit = 0,
    i = 1,
    len = prices.length;

  while (i < len) {
    while (i < len && prices[i - 1] <= prices[i]) {
      i++;
    }
    const buy = prices[i - 1];
    while (i < len && prices[i - 1] >= prices[i]) {
      i++;
    }
    const sale = prices[i - 1];
    profit += sale - buy;
  }
  return profit;
};

File: 122/py/solution.py
class Solution:
    def maxProfit(self, prices):
        if len(prices) <= 1:
            return 0

        profit = 0
        for i in range(1, len(prices)) :
            if prices[i] - prices[i - 1] > 0 :
                profit += prices[i] - prices[i - 1]
            
        return profit
